src/components/NewTaskForm.js

```diff
@@ -1,44 +1,50 @@
 import React, { useState } from "react";
 
-function NewTaskForm({ categories, onTaskFormSubmit, onDetails, handleChange }) {
-const [taskCategory, setTaskCategory] = useState("Code")
+function NewTaskForm({ categories, onTaskFormSubmit }) {
+  const [taskCategory, setTaskCategory] = useState("Code");
+  const [onDetails, setOnDetails] = useState("");
 
-  const options = categories.map(cats => {
+  const options = categories.map((cats) => {
     if (cats !== "All") {
-      return <option key={cats} value={cats}>{cats}</option>
+      return (
+        <option key={cats} value={cats}>
+          {cats}
+        </option>
+      );
     }
-    return cats
- })
+    return cats;
+  });
 
-function updateCategory(e){
-  setTaskCategory(e.target.value)
-}
-
- function handleClick(e){
-  e.preventDefault()
-  // if (onDetails.length >= 1) {
-    const newItem = {
+  function handleSubmit(e) {
+    e.preventDefault();
+    onTaskFormSubmit({
       text: onDetails,
       category: taskCategory,
-    }
-    onTaskFormSubmit(newItem)
-  // }
- }
-
+    });
+  }
 
   return (
-    <form className="new-task-form">
+    <form className="new-task-form" onSubmit={handleSubmit}>
```

Comment:
form is submitting preventing new items from being added under test. Moving the handler here fixes it